### PR TITLE
Added remaining tests for full covering _optimizer.py

### PR DESCRIPTION
**Pull request by PredictiveManish**
## Overview

Working towards #3690  

Description of the changes proposed in this pull request:  
This PR adds 7 new test cases to improve test coverage for `shap/actions/_optimizer.py`, addressing the low coverage (15%) identified in the testing roadmap.

## Tests Added:

1. `test_warning_emitted` - Verifies that the alpha state warning is emitted when ActionOptimizer is initialized.
2. `test_no_valid_solution` - Tests the edge case when no valid solution exists and the queue empties without finding a solution.
3. `test_action_replacement_in_group` - Tests the core logic of replacing an action with a more expensive one in the same action group.
4. `test_multiple_action_groups_with_replacement` - Tests combination scenarios with multiple action groups where actions are replaced within the same group.
5. `test_immediate_satisfaction` - Tests the edge case when the model is satisfied on the first try without needing additional actions.
6. `test_single_action_group` - Tests the case when actions contains a single Action object (not wrapped in a list).
7. `test_action_list_sorting` - Verifies that actions provided in a list are properly sorted by cost.

## Coverage Impact

These tests significantly improve coverage of `shap/actions/_optimizer.py`, particularly around:

- Warning emission on initialization  
- Queue handling and action replacement logic  
- Edge cases for solution finding  
- Action sorting and grouping behavior  

Note: This file will be renamed to `test_optimzer.py` after #4350 is merged.

## Checklist

- [x] All [pre-commit checks](https://pre-commit.com/#install) pass.
- [x] Unit tests added (if fixing a bug or adding a new feature)

### CODE REVIEW

File: tests/actions/_optimizer.py [hunk 1]
```diff
@@ -93,3 +93,115 @@ def test_run_out_of_group():
 def test_bad_action():
     with pytest.raises(InvalidAction):
         shap.ActionOptimizer(None, [None])  # type: ignore
+
+
+def test_warning_emitted():
```
Comment (CloseChoice) [inline tests/actions/_optimizer.py:98]: does this test add any value? I don't really see that.

File: tests/actions/_optimizer.py [hunk 2]
```diff
@@ -93,3 +93,115 @@ def test_run_out_of_group():
...
+        [IncreaseFeature2(i) for i in range(1, 5)],
+    ]
+    optimizer = shap.ActionOptimizer(sum_equals_12, possible_actions)
+    actions = optimizer(X.iloc[0])
+    assert len(actions) == 2
+
+
+def test_immediate_satisfaction():
+    """Tests when model is satisfied on first try (no actions needed)."""
+    X, IncreaseFeature1, IncreaseFeature2, IncreaseFeature3, passed = create_basic_scenario()
```
Comment (CloseChoice) [inline tests/actions/_optimizer.py:161]: why not 

`X, IncreaseFeature1, _, _, _ = ...`?

**Review by CloseChoice — CHANGES_REQUESTED** [review]
Pls resolve the merge conflict, so that these tests are actually run!

File: tests/actions/_optimizer.py (hunk 1 above)
Comment (PredictiveManish) [reply tests/actions/_optimizer.py]: yes, i think too, added previously to ensure the warning isn't accidentally removed, but it does feel like testing implementation details rather than user-facing behavior.

File: tests/actions/_optimizer.py (hunk 2 above)
Comment (PredictiveManish) [reply tests/actions/_optimizer.py]: yes, missed that.

**Comment by PredictiveManish** [comment]
@CloseChoice Resolved the conflict and updated tests.
